### PR TITLE
[JetStream] Update tv-material to 1.0.0 stable

### PR DESCRIPTION
**Pull request by chikoski**
This PR updates tv-material library to 1.0.0 stable. 

In addition to that, following changes are also applied:

- Some dependencies also updated 
- Target SDK is updated to 35
- Removal of experimental flags

### CODE REVIEW

File: JetStreamCompose/gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,40 +1,34 @@
 [versions]
-activity-compose = "1.9.0"
-android-gradle-plugin = "8.4.1"
-android-test-plugin = "8.4.1"
-androidx-baselineprofile = "1.2.4"
-benchmark-macro-junit4 = "1.2.4"
+activity-compose = "1.9.2"
+android-gradle-plugin = "8.6.0"
```
Comment (mlykotom) [inline JetStreamCompose/gradle/libs.versions.toml:3]: nit, but you use test plugin newer version below this line :)

```suggestion
android-gradle-plugin = "8.6.1"
```

File: JetStreamCompose/gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
+android-test-plugin = "8.6.1"
+androidx-baselineprofile = "1.3.0"
+benchmark-macro-junit4 = "1.3.0"
 coil-compose = "2.6.0"
-compose-bom = "2024.05.00"
-compose-foundation = "1.7.0-beta02"
-compose-ui = "1.7.0-beta02"
-compose-runtime = "1.7.0-beta02"
-tv-material = "1.0.0-beta01"
+compose-bom = "2024.09.01"
```
Comment (mlykotom) [inline JetStreamCompose/gradle/libs.versions.toml:8]:
```suggestion
compose-bom = "2024.09.02"
```

**Review by IanGClifton — COMMENTED** [review]
Can this also address https://github.com/android/tv-samples/blob/main/JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/utils/BringIntoViewIfChildrenAreFocused.kt ? Otherwise LGTM with Tomas' feedback

**Comment by chikoski** [comment]
Removed an annotation from https://github.com/android/tv-samples/blob/main/JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/utils/BringIntoViewIfChildrenAreFocused.kt

I would defer the Modifier.Node migration to another PR.

**Pull request merged**